### PR TITLE
Remove 3DNow! from X86TargetParser.

### PR DESCRIPTION
**Pull request by jyknight**
This addresses the spurious inclusion of (now unsupported) target features '-3dnow' and '-3dnowa' when disabling mmx (when then caused log output from `clang -mno-mmx`).

It should've been part of PR #96246, but was missed.

Also tweaks the warning in prfchwintrin.h to not recommend the deprecated mm3dnow.h header.

### CODE REVIEW

File: llvm/lib/TargetParser/X86TargetParser.cpp [hunk 1]
```diff
@@ -171,14 +171,14 @@ constexpr FeatureBitset FeaturesClearwaterforest =
 
 // Geode Processor.
 constexpr FeatureBitset FeaturesGeode =
-    FeatureX87 | FeatureCMPXCHG8B | FeatureMMX | Feature3DNOW | Feature3DNOWA;
```
Comment (RKSimon) [inline llvm/lib/TargetParser/X86TargetParser.cpp:174]: Do we need to add a FeaturePREFTECH of some kind?

Comment (jyknight) [reply llvm/lib/TargetParser/X86TargetParser.cpp:174]: Possibly not?

We didn't enable it before (so, e.g. `__PRFCHW__` isn't and wasn't defined), and while we DID enable the prefetchw instruction in the backend for these targets, we still do now (via implying it in X86.td).

OTOH, maybe it's weird to have that divergence, so I'd also be fine with adding it (and therefore setting the preprocessor define).

Comment (jyknight) [reply llvm/lib/TargetParser/X86TargetParser.cpp:174]: (I somewhat lean towards submitting the commit as-is, but will change this if you wish)

Comment (RKSimon) [reply llvm/lib/TargetParser/X86TargetParser.cpp:174]: I think we should add the define. If/when we eventually remove mm3dnow.h then the only way to access prfchwintrin.h is via x86intrin.h (which we already recommend) which can sometimes require the `__PRFCHW__` define:
https://github.com/llvm/llvm-project/blob/f6add66b720f85bf1092af7d6702b7397da57349/clang/lib/Headers/x86intrin.h#L17-L19

Also, we still have references to mm3dnow.h in prfchwintrin.h that probably need removing:
https://github.com/llvm/llvm-project/blob/f6add66b720f85bf1092af7d6702b7397da57349/clang/lib/Headers/prfchwintrin.h#L10-L12

Comment (jyknight) [reply llvm/lib/TargetParser/X86TargetParser.cpp:174]: OK, making the change.

I note that "sometimes" is essentially "if `__SCE__` is defined". Really weird to have that be Sony-specific behavior.

File: llvm/include/llvm/TargetParser/X86TargetParser.def [hunk 2]
```diff
@@ -175,8 +175,6 @@ X86_FEATURE_COMPAT(AVX512BF16,      "avx512bf16",            34)
 X86_FEATURE_COMPAT(AVX512VP2INTERSECT, "avx512vp2intersect", 35)
 // Below Features has some missings comparing to gcc, it's because gcc has some
 // not one-to-one mapped in llvm.
-X86_FEATURE_COMPAT(3DNOW,           "3dnow",                  0)
-X86_FEATURE       (3DNOWA,          "3dnowa")
```
Comment (phoebewang) [inline llvm/include/llvm/TargetParser/X86TargetParser.def:179]: @FreddyLeaf IIRC, we need to find two features to fill here, right?

Comment (FreddyLeaf) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: Yes, if gcc also deleted 3dnow, then find 1 feature to fill here. if not, then find 2.

Comment (FreddyLeaf) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: Sorry, I realized even gcc removed, they will keep one blank here to keep backward compatibility. So two features are required.
according to this comment above the file: 
```
// We cannot re-adjust the position of X86_FEATURE_COMPAT at the whole list.

Comment (KanRobert) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: I think you can put sub-features of APX here if you do not have better candidates. `egpr,ndd,push2pop2,ppx,ccmp,cf,nf,zu`

Comment (FreddyLeaf) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: A bad news is that only X86_`zu` can be "used" here. all other sub-features of APX have already be "used."

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

File: llvm/include/llvm/TargetParser/X86TargetParser.def (hunk 2 above)
Comment (jyknight) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: I don't understand this thread. What exactly needs to match what else?

Also, if it's important for each feature to be assigned a specific ABI-stable number for some reason, we should assign that number _directly_ in this macro, not indirectly via ordering...

Comment (jyknight) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: Could I just put this in place of these two lines for now?
```
X86_FEATURE       (DUMMYFEATURE1,           "__dummyfeature1")
X86_FEATURE       (DUMMYFEATURE2,          "__dummyfeature2")
```

Comment (jyknight) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: I've gone ahead and done that.

**Review by phoebewang — APPROVED** [review]
LGTM.

File: llvm/include/llvm/TargetParser/X86TargetParser.def (hunk 2 above)
Comment (phoebewang) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:179]: Maybe remove here too https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/builtins/cpu_model/x86.c#L144

**Pull request merged**